app/backend/src/couchers/servicers/blocking.py

```diff
@@ -72,17 +72,23 @@ def UnblockUser(self, request, context, session):
         return empty_pb2.Empty()
 
     def GetBlockedUsers(self, request, context, session):
-        blocked_users = (
-            session.execute(
-                select(User)
-                .join(UserBlock, UserBlock.blocked_user_id == User.id)
-                .where(User.is_visible)
-                .where(UserBlock.blocking_user_id == context.user_id)
-            )
-            .scalars()
-            .all()
-        )
+        blocked_users = session.execute(
+            select(User.username, User.name, Upload.filename)
+            .join(UserBlock, UserBlock.blocked_user_id == User.id)
+            .outerjoin(Upload, Upload.key == User.avatar_key)
+            .where(User.is_visible)
+            .where(UserBlock.blocking_user_id == context.user_id)
+        ).all()
 
         return blocking_pb2.GetBlockedUsersRes(
-            blocked_usernames=[blocked_user.username for blocked_user in blocked_users],
+            blocked_users=[
+                blocking_pb2.BlockedUser(
+                    username=blocked_user.username,
+                    name=blocked_user.name,
+                    avatar_thumbnail_url=Upload(filename=blocked_user.filename).thumbnail_url
```

Comment:
Replace with `urls.media_url("thumbnail", blocked_user.filename)`

Reply:
Wait which part? Can you do a suggest changes?